In [22]:
import pandas as pd
from sklearn import preprocessing
import warnings
warnings.filterwarnings('ignore')
from sklearn.model_selection import train_test_split
from sklearn.naive_bayes import GaussianNB
from sklearn.metrics import confusion_matrix
from sklearn.metrics import accuracy_score

In [24]:
df=pd.read_csv('SVMtrain.csv')

In [12]:
df.head()

PassengerId  Survived  Pclass     Sex   Age  SibSp  Parch     Fare  \
0            1         0       3    Male  22.0      1      0   7.2500   
1            2         1       1  female  38.0      1      0  71.2833   
2            3         1       3  female  26.0      0      0   7.9250   
3            4         1       1  female  35.0      1      0  53.1000   
4            5         0       3    Male  35.0      0      0   8.0500   

   Embarked  
0         3  
1         1  
2         3  
3         3  
4         3

In [13]:
df.columns

Index(['PassengerId', 'Survived', 'Pclass', 'Sex', 'Age', 'SibSp', 'Parch',
       'Fare', 'Embarked'],
      dtype='object')

In [25]:
from sklearn import preprocessing
from sklearn.preprocessing import LabelEncoder
le=preprocessing.LabelEncoder()
le.fit(df["Sex"])
LabelEncoder()
print(le.classes_)
df["Sex"]=le.transform(df["Sex"])

['Male' 'female']


In [27]:
df1=df.drop(['PassengerId'],axis=1)

In [28]:
df1

Survived  Pclass  Sex   Age  SibSp  Parch     Fare  Embarked
0           0       3    0  22.0      1      0   7.2500         3
1           1       1    1  38.0      1      0  71.2833         1
2           1       3    1  26.0      0      0   7.9250         3
3           1       1    1  35.0      1      0  53.1000         3
4           0       3    0  35.0      0      0   8.0500         3
..        ...     ...  ...   ...    ...    ...      ...       ...
884         0       2    0  27.0      0      0  13.0000         3
885         1       1    1  19.0      0      0  30.0000         3
886         0       3    1  60.0      1      2  23.4500         3
887         1       1    0  26.0      0      0  30.0000         1
888         0       3    0  32.0      0      0   7.7500         2

[889 rows x 8 columns]

In [30]:
y=df1['Survived']

In [31]:
x=df1.drop(["Survived"],axis=1)

In [32]:
x.head()

Pclass  Sex   Age  SibSp  Parch     Fare  Embarked
0       3    0  22.0      1      0   7.2500         3
1       1    1  38.0      1      0  71.2833         1
2       3    1  26.0      0      0   7.9250         3
3       1    1  35.0      1      0  53.1000         3
4       3    0  35.0      0      0   8.0500         3

In [36]:
x_train,x_test,y_train,y_test=train_test_split(x,y,test_size=0.3,random_state=True)

In [37]:
x_train.head()

Pclass  Sex   Age  SibSp  Parch     Fare  Embarked
114       3    0  21.0      0      0   7.9250         3
872       2    1  28.0      1      0  24.0000         1
76        3    0  60.0      0      0   8.0500         3
874       3    0  20.0      0      0   9.8458         3
681       3    0  20.0      0      0   9.2250         3

In [38]:
x_test.head()

Pclass  Sex   Age  SibSp  Parch     Fare  Embarked
386       2    1  36.0      0      0  13.0000         3
258       2    1  50.0      0      1  26.0000         3
770       3    0  48.0      0      0   7.8542         3
207       3    1  16.0      0      0   7.7500         2
682       3    0  14.0      5      2  46.9000         3

In [39]:
y_train.head()

114    0
872    1
76     0
874    0
681    0
Name: Survived, dtype: int64

In [40]:
y_test.head()

386    1
258    1
770    0
207    1
682    0
Name: Survived, dtype: int64

In [43]:
#testdata
0.3*889

266.7

In [44]:
#trainingdata
0.7*889

622.3

In [46]:
from sklearn.naive_bayes import *
clf=BernoulliNB()
y_pred=clf.fit(x_train,y_train).predict(x_test)
accuracy_score(y_test,y_pred,normalize=True)

0.8202247191011236

In [47]:
confusion_matrix(y_test,y_pred)

array([[144,  22],
       [ 26,  75]], dtype=int64)

In [48]:
y=df1['Pclass']

In [49]:
x=df1.drop(['Pclass'],axis=1)

In [50]:
x.head()

Survived  Sex   Age  SibSp  Parch     Fare  Embarked
0         0    0  22.0      1      0   7.2500         3
1         1    1  38.0      1      0  71.2833         1
2         1    1  26.0      0      0   7.9250         3
3         1    1  35.0      1      0  53.1000         3
4         0    0  35.0      0      0   8.0500         3

In [51]:
x_train,x_test,y_train,y_test=train_test_split(x,y,test_size=0.3,random_state=True)

In [53]:
clf=BernoulliNB()
y_pred=clf.fit(x_train,y_train).predict(x_test)
accuracy_score(y_test,y_pred,normalize=True)

0.5805243445692884

In [54]:
confusion_matrix(y_test,y_pred)

array([[ 37,   0,  34],
       [ 20,   0,  33],
       [ 25,   0, 118]], dtype=int64)

In [61]:
y=df1['Sex']

In [62]:
x=df1.drop(['Sex'],axis=1)

In [63]:
x_train,x_test,y_train,y_test=train_test_split(x,y,test_size=0.3,random_state=True)

In [64]:
clf=BernoulliNB()
y_pred=clf.fit(x_train,y_train).predict(x_test)
accuracy_score(y_test,y_pred,normalize=True)

0.7453183520599251

In [65]:
confusion_matrix(y_test,y_pred)

array([[160,  10],
       [ 58,  39]], dtype=int64)

In [69]:
y=df1['Embarked']
x=df1.drop(['Embarked'],axis=1)
x_train,x_test,y_train,y_test=train_test_split(x,y,test_size=0.3,random_state=True)
clf=BernoulliNB()
y_pred=clf.fit(x_train,y_train).predict(x_test)
accuracy_score(y_test,y_pred)

0.700374531835206

In [70]:
confusion_matrix(y_test,y_pred)

array([[  0,   0,  54],
       [  0,   0,  26],
       [  0,   0, 187]], dtype=int64)

In [71]:
y=df1['SibSp']
x=df1.drop(['SibSp'],axis=1)
x_train,x_test,y_train,y_test=train_test_split(x,y,test_size=0.3,random_state=True)
clf=BernoulliNB()
y_pred=clf.fit(x_train,y_train).predict(x_test)
accuracy_score(y_test,y_pred)

0.6554307116104869

In [72]:
confusion_matrix(y_test,y_pred)

array([[162,  16,   0,   0,   0,   0,   0],
       [ 47,  13,   0,   0,   0,   0,   0],
       [  6,   6,   0,   0,   0,   0,   0],
       [  3,   3,   0,   0,   0,   0,   0],
       [  6,   1,   0,   0,   0,   0,   0],
       [  3,   0,   0,   0,   0,   0,   0],
       [  0,   1,   0,   0,   0,   0,   0]], dtype=int64)

In [73]:
y=df1['Parch']
x=df1.drop(['Parch'],axis=1)
x_train,x_test,y_train,y_test=train_test_split(x,y,test_size=0.3,random_state=True)
clf=BernoulliNB()
y_pred=clf.fit(x_train,y_train).predict(x_test)
accuracy_score(y_test,y_pred)

0.7191011235955056

In [74]:
confusion_matrix(y_test,y_pred)

array([[184,  17,   0,   0,   0,   0],
       [ 34,   8,   0,   0,   0,   0],
       [ 19,   2,   0,   0,   0,   0],
       [  1,   0,   0,   0,   0,   0],
       [  1,   0,   0,   0,   0,   0],
       [  1,   0,   0,   0,   0,   0]], dtype=int64)

so, when the dependent column was Survived thw model has the greatest accuracy that 82%